src/context/filter.rs

```diff
@@ -0,0 +1,155 @@
+use crate::{
+    Context, RedisModuleCommandFilter, RedisModuleCommandFilterCtx, RedisModuleString,
+    RedisModule_CommandFilterArgDelete, RedisModule_CommandFilterArgGet,
+    RedisModule_CommandFilterArgInsert, RedisModule_CommandFilterArgReplace,
+    RedisModule_CommandFilterArgsCount, RedisModule_CommandFilterGetClientId,
+    RedisModule_CreateString, RedisModule_RegisterCommandFilter,
+    RedisModule_UnregisterCommandFilter, ValkeyString,
+};
+use std::ffi::{c_int, CString};
+use std::ptr::null_mut;
+use std::str::Utf8Error;
+
+#[derive(Debug, Clone, Copy)]
+pub struct CommandFilter {
+    inner: *mut RedisModuleCommandFilter,
+}
+
+// otherwise you get error:     cannot be shared between threads safely
+unsafe impl Send for CommandFilter {}
+unsafe impl Sync for CommandFilter {}
+
+/// CommandFilter is a wrapper around the RedisModuleCommandFilter pointer
+///
+/// It provides a way to check if the filter is null and to create a new CommandFilter
+impl CommandFilter {
+    pub fn new(inner: *mut RedisModuleCommandFilter) -> Self {
+        CommandFilter { inner }
+    }
+
+    pub fn is_null(&self) -> bool {
+        self.inner.is_null()
+    }
+}
+
+pub struct CommandFilterCtx {
+    inner: *mut RedisModuleCommandFilterCtx,
+}
+
+/// wrapping the RedisModuleCommandFilterCtx to provide a higher level interface to call RedisModule_CommandFilter*
+///
+/// provides functions to interact with the command filter context, such as getting the number of arguments, getting and replacing arguments, and deleting arguments.
+impl CommandFilterCtx {
+    pub fn new(inner: *mut RedisModuleCommandFilterCtx) -> Self {
+        CommandFilterCtx { inner }
+    }
+
+    /// wrapper for RedisModule_CommandFilterArgsCount
+    pub fn args_count(&self) -> c_int {
+        unsafe { RedisModule_CommandFilterArgsCount.unwrap()(self.inner) }
+    }
+
+    /// wrapper for RedisModule_CommandFilterArgGet
+    pub fn arg_get(&self, pos: c_int) -> *mut RedisModuleString {
```

Comment:
Could we update this API to return a Rust like structure - &str, String, or ValkeyString - instead of a raw pointer?

I would suggest using `ValkeyString::from_redis_module_string()` for this

Reply:
added `arg_get_try_as_str` function